netket/experimental/sampler/metropolis_pt.py

```diff
@@ -344,28 +373,16 @@ def fix_swap(do_swap, swap_order):
 
             # Update statistics to compute diffusion coefficient of replicas
             # Total exchange steps performed
+            s["exchange_steps"] += 1
```

Comment:
do we need to update this number in the loop? after every loop you can directly update it by n_sweeps I think?

Reply:
I put the update here only so that we directly access this quantity in the calculation of the diffusion. If you prefer this to be updated after the loop, i can change the diffusion correctly from `s["beta_position"] = s["beta_position"] + delta / s["exchange_steps"]' to `s["beta_position"] = s["beta_position"] + delta / (state.exchange_steps + jnp.asarray(i, dtype=jnp.int64) + 1)'. Both work for me

Reply:
I'm not sure I understand. Here you are just adding 1 in every step of the loop, meaning that after n sweep iterations you will have added n-sweep...

No?

Reply:
Yes indeed, the final value is the same. 
But inside the loop, I use the updated value of the variable to calculate `s["beta_position"]`. One simple solution would be to calculate the denominator directly, instead of accessing the variable. What I mean exactly : 

- either we keep it this way : 
```python
  s["exchange_steps"] += 1
  s["beta_position"] = s["beta_position"] + delta / s["exchange_steps"]
(...)
new_state = state.replace(
    exchange_steps=s["exchange_steps"],
)
```

- or we don't update this variable inside the loop
```python
  s["beta_position"] = s["beta_position"] + delta / (state.exchange_steps + jnp.asarray(i+1, dtype=jnp.int64) )
(...)
new_state = state.replace(
    exchange_steps=state.exchange_steps + sampler.sweep_size
)
```
In my opinion, updating it inside the loop just makes it a bit clearer for the calculation of `beta_position`, but it doesn't change so much. I can do the second way if you think it is more efficient.

Reply:
It's ok, it's not very important given everything else going on.
Let's keep it that way.